src/Tiny_httpd.ml

```diff
@@ -83,6 +83,27 @@ module Byte_stream = struct
   let of_chan = of_chan_ ~close:close_in
   let of_chan_close_noerr = of_chan_ ~close:close_in_noerr
 
+  exception Timeout
+
+  let of_descr_ ?(timeout=(-1.0)) ~close ic : t =
+    let i = ref 0 in
+    let len = ref 0 in
+    let buf = Bytes.make 4096 ' ' in
```

Comment:
bigger buffers are probably nice, like 16 kb at least :). Not your fault, I used this value before.

Reply:
Maybe we can make a new parameter to create. A server handling many small request will prefer smaller buffer than 
a server handling big, but less many request. So it is nive to let the user choose ?

I also thought using your buffered input above ocaml's channel was stacking buffer ... So my PR also addresses that.

Reply:
in_channel use a large buffer (64Kb from memory) and is written in C, I am not very surprised... It is not slower at least ;-)